LeaderboardBackend/Services/Impl/RunService.cs

```diff
@@ -11,6 +12,49 @@ public class RunService(ApplicationContext applicationContext) : IRunService
     public async Task<Run?> GetRun(Guid id) =>
         await applicationContext.Runs.Include(run => run.Category).SingleOrDefaultAsync(run => run.Id == id);
 
+    public async Task<GetRunsForCategoryResult> GetRunsForCategory(
+        long id,
+        Page page,
+        bool includeDeleted = false
+    )
+    {
+        Category? cat = await applicationContext.FindAsync<Category>(id);
+        if (cat is null)
+        {
+            return new NotFound();
+        }
+
+        if (cat.DeletedAt is not null)
+        {
+            return new AlreadyDeleted();
+        }
+
+        IQueryable<Run> query = applicationContext.Runs
+            .Include(run => run.Category)
+            .Where(run =>
+                run.CategoryId == id && (includeDeleted || run.DeletedAt == null)
+            );
+
+        long count = await query.LongCountAsync();
+
+        if (cat.SortDirection == SortDirection.Descending)
+        {
+            query = query.OrderByDescending(run => run.TimeOrScore)
+                .ThenBy(run => run.PlayedOn);
+        }
+        else
+        {
+            query = query.OrderBy(run => run.TimeOrScore)
+                .ThenBy(run => run.PlayedOn);
+        }
```

Comment:
We should probably have a third tiebreaker such as creation date or something.

Reply:
Adding a TODO at the top of this block stating that ultimately we should fetch ranking data from Redis.